src/app/services/Document.service.ts

```diff
@@ -167,14 +167,44 @@ export class DocumentService {
     return this.documentMap.toDTO(document);
   }
 
-  async search(query: Partial<Document>): Promise<IDocumentOutgoingDTO[]> {
-    const elasticResults = await this.elasticDocumentRepository.search(query);
-
-    const documents = elasticResults.map((result) => {
-      const props = this.documentMap.toDocumentProps(result);
-      return new Document(props);
-    });
+  async search(query: ElasticSearchProps): Promise<IDocumentOutgoingDTO[]> {
+    const nonEmptyQuery = Object.keys(query)
+      .filter((key) => query[<keyof ElasticSearchProps>key] !== '')
+      .reduce((docObj: Record<string, any>, key: string) => {
+        docObj[key] = query[<keyof ElasticSearchProps>key];
+        return docObj;
+      }, {});
+
+    const elasticResults = await this.elasticDocumentRepository.search(nonEmptyQuery);
+
+    const documents: IDocumentOutgoingDTO[] = [];
+
+    for (let i = 0; i < elasticResults.length; i++) {
```

Comment:
I'd replace this for with a Promise.allSettled(elasticResults.map( // logic here))